runtime/java_test.go

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (C) 2017 XLAB, Ltd.
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+package runtime
+
+import (
+	"testing"
+
+	. "github.com/mikelangelo-project/capstan/testing"
+	. "gopkg.in/check.v1"
+	"gopkg.in/yaml.v2"
+)
+
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type javaSuite struct {
+}
+
+var _ = Suite(&javaSuite{})
+
+func (*javaSuite) TestGetBootCmd(c *C) {
+	// Simulate openjdk8-zulu-compact1's meta/run.yaml being parsed.
+	cmdConfs := map[string]*CmdConfig{
+		"openjdk8-zulu-compact1": &CmdConfig{
+			RuntimeType:      Native,
+			ConfigSetDefault: "java",
+			ConfigSets: map[string]Runtime{
+				"java": nativeRuntime{
+					BootCmd: "/java.so",
+					CommonRuntime: CommonRuntime{
+						Env: map[string]string{
+							"XMS":       "512m",
```

Comment:
I was thinking: in unit tests we could theoretically either pull whole packages from S3 repository, either only meta/run.yamls from capstan-packages recipe. Both should generally be synced, but recall how there is usually quite some delay before we upload updated recipes to the S3.

So I'm wondering wheter Capstan unit tests should really struggle with actual packages or should we only make sure that bootcmd generation works as expected. I would somehow prefer if no network traffic was produced during unit tests. Shall we sleep over and decide tomorrow?